src/Interpreters/TemporaryDataOnDisk.cpp

```diff
@@ -65,7 +65,7 @@ TemporaryDataOnDisk::TemporaryDataOnDisk(TemporaryDataOnDiskScopePtr parent_, Cu
 
 std::unique_ptr<WriteBufferFromFileBase> TemporaryDataOnDisk::createRawStream(size_t max_file_size)
 {
-    if (file_cache)
+    if (file_cache && file_cache->isInitialized())
```

Comment:
This might be a breaking change for those who rely only on cache for temporary data and do not have volume for tmp data defined in configuration... Let's disable async cache loading by default for now and add randomization of this setting for tests.

Reply:
> Let's disable async cache loading by default for now

Yes, I just want to make sure the CI passes with it enabled by default, as per this comment https://github.com/ClickHouse/ClickHouse/pull/65736#discussion_r1699837710. Once approved, I'll revert to disabling it by default.

Reply:
Disabled by default